app/services/username_generator_service.rb

```
@@ -37,7 +37,15 @@ def ensure_valid_username(name)
       name += "a" if name.blank? || name.match?(/^[0-9]+$/)
       name += random_digit if DENYLIST.include?(name)
       name += random_digit while name.length < 3
-      name.first(20)
+
+      # If the name is purely numeric and longer than 19 characters, truncate to 19
+      # to reserve space for the letter that will be added
+      if name.match?(/^[0-9]+$/) && name.length > 19
+        name = name.first(19)
+        name += "a"
```

RuboCop (1.75.5)

[warning] Useless assignment to variable - `name`.

(Lint/UselessAssignment)